.circleci/config.yml

```diff
@@ -53,11 +53,11 @@ jobs:
           command: |
             . venv/bin/activate
             pep8 --ignore=E402 webcompat/ tests/ config/secrets.py.example
-            python run.py -t &
-            sleep 5
-            nosetests
             npm run lint
             npm run build
+            nosetests
```

Comment:
@marimeireles did you have an error message? What was it?

Reply:
@karlcow 
You can check the log here: https://circleci.com/gh/webcompat/webcompat.com/363?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
I've just updated the repo with the code you told me to use.

Reply:
Thanks a lot! I will check.

Reply:
ah! Understood!

```
  File "/home/circleci/repo/webcompat/helpers.py", line 532, in policy
    response = make_response(view(*args, **kwargs))
  File "/home/circleci/repo/webcompat/views.py", line 321, in show_issue
    push(bust_cache('/js/dist/webcompat.min.js'), **{
  File "/home/circleci/repo/webcompat/helpers.py", line 61, in bust_cache
    return file_path + '?' + get_checksum(STATIC_PATH + file_path)
  File "/home/circleci/repo/webcompat/helpers.py", line 57, in get_checksum
    checksum = md5_checksum(file_path)
  File "/home/circleci/repo/webcompat/helpers.py", line 66, in md5_checksum
    with open(file_path, 'rb') as fh:
IOError: [Errno 2] No such file or directory: '/home/circleci/repo/webcompat/static/js/dist/webcompat.min.js'
```

This is happening because the test relies now  on this 

```
push(bust_cache('/js/dist/webcompat.min.js'), **{
```

AND this 

```
return file_path + '?' + get_checksum(STATIC_PATH + file_path)
```

which indeed will create an issue if `'/js/dist/webcompat.min.js'` has not been generated.

I need to dive in the test also look if it might create issues when running on localhost.

Reply:
There's another one which is slightly different and to be checked. 

```
======================================================================
FAIL: Page title format for different URIs.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/repo/tests/unit/test_rendering.py", line 66, in test_titles
    self.assertTrue(expected in rv.data)
  File "/home/circleci/repo/venv/lib/python2.7/site-packages/flask/ctx.py", line 243, in __exit__
    self.pop(exc_value)
  File "/home/circleci/repo/venv/lib/python2.7/site-packages/flask/ctx.py", line 235, in pop
    % (rv, self)
AssertionError: Popped wrong app context.  (<flask.ctx.AppContext object at 0x7fd8d6222350> instead of <flask.ctx.AppContext object at 0x7fd8d61e2c10>)
```

Reply:
That said there is something interesting.

pulling out @marimeireles branch

```
→ git checkout -b marimeireles-issues/2778/1 master
Switched to a new branch 'marimeireles-issues/2778/1'
🐝 14:28:33 ~/code/webcompat.com
→ git pull https://github.com/marimeireles/webcompat.com.git issues/2778/1
remote: Enumerating objects: 13, done.
remote: Counting objects: 100% (13/13), done.
remote: Total 17 (delta 11), reused 13 (delta 11), pack-reused 4
Unpacking objects: 100% (17/17), done.
From https://github.com/marimeireles/webcompat.com
 * branch            issues/2778/1 -> FETCH_HEAD
Merge made by the 'recursive' strategy.
 .circleci/config.yml    |  5 ++---
 config/requirements.txt |  1 +
 webcompat/__init__.py   |  5 +++++
 webcompat/views.py      | 50 ++++++++++++++++++++++++++++++++++++++++++++++++++
 4 files changed, 58 insertions(+), 3 deletions(-)
```

Then checking the tests.

```
🐝 14:28:54 ~/code/webcompat.com
→ nosetests
.............................................................................................................
----------------------------------------------------------------------
Ran 109 tests in 2.437s

OK
```

Everything runs without issue.

It should not fail on CircleCI too.

I'm not an expert in CircleCI config. but I noticed that we activate twice the environment?

Here
https://github.com/webcompat/webcompat.com/blob/master/.circleci/config.yml#L31

and here
https://github.com/webcompat/webcompat.com/blob/master/.circleci/config.yml#L54

But this should not matter for nosetests.

This branch is:

```
      # run tests!
      - run:
          name: run tests
          command: |
            . venv/bin/activate
            pep8 --ignore=E402 webcompat/ tests/ config/secrets.py.example
            nosetests
            npm run lint
            npm run build
            python run.py -t &
            sleep 5
            node --max-old-space-size=8192 ./tests/functional/_intern.js --reporters="runner" --firefoxBinary=`which firefox`
            firefox --version
```

Reply:
Gotcha!

Partly Flawed tests.
Partly the modification of @marimeireles 

And a good lesson to learn and improve the tests.

Currently when we run nosetests on a localhost, we usually do it without erasing the static files.

Let me do that and run the tests.

```
🐝 14:49:28 ~/code/webcompat.com
→ rm -rf webcompat/static/js/*.js
🐝 14:49:53 ~/code/webcompat.com
→ rm -rf webcompat/static/js/build/
🐝 14:50:01 ~/code/webcompat.com
→ rm -rf webcompat/static/js/dist/
🐝 14:50:07 ~/code/webcompat.com
→ rm -rf webcompat/**/*.min.css
🐝 14:50:14 ~/code/webcompat.com
→ rm -rf webcompat/static/css/dist/
🐝 14:50:20 ~/code/webcompat.com
→ nosetests
.............................E.................EEEEEFE.........................EEEEEEEEEE..E.................
```

ooopsy…

OK let's go back to master.

```
🐝 14:50:25 ~/code/webcompat.com
→ git checkout master
Switched to branch 'master'
Your branch is up-to-date with 'webcompat/master'.

```
Then let's make sure the static files are not here.

```
🐝 14:56:12 ~/code/webcompat.com
→ find . -name webcompat.min.js
```

ok nothing

then let's run the tests.

```
🐝 14:56:41 ~/code/webcompat.com
→ nosetests
.............................................................................................................
----------------------------------------------------------------------
Ran 109 tests in 1.166s

OK
```

working! no flawed tests.

So the results of the modifications by @marimeireles is that it broke tests because it pushes a resource which is not available when doing nosetests. Which means we included a new dependency that should not exist.

There are probably different ways to solve this. 

For example to use the push feature only when we are not in LOCALHOST.
OR to mock the files.

But these could be the topic of a new issue.

Reply:
@marimeireles has a summary.

You can move back nosetests after `npm run build` in the meantime. And we will find a solution later for it where the unit tests will not be dependent of building the secondary resources.

THANKS!

Reply:
Ok!
Thanks for the clarifying review @karlcow. 
Do you want me to file an issue?

Reply:
@marimeireles filing an issue would be good. yup. Thanks. "Make sure unittest are not dependent of compiled assets"